### ISSUE TITLE
[Help] Running help on an invalid command raises errors in DM with custom checks

### ISSUE THREAD

**Issue by TrustyJAID** [Cog-Creators/Red-DiscordBot#4390]
# Command bugs

#### Command name

`[p]help`

#### What cog is this command from?

Help formatter

#### What were you expecting to happen?

`[p]help non_existant_command_or_group` returns nothing

#### What actually happened?

`[p]help non_existant_command_or_group` raises an error unrelated to anything in particular

#### How can we reproduce this issue?

Run `[p]help <some command that doesn't exist>` in DM with the bot with custom build checks.

**Comment by Jackenmen** [comment]
I can't reproduce this, I would need a bit more details than what's presented here. Output of `[p]helpset showsettings` as well as the traceback (and code related to it, if it's 3rd-party) could help here.

**Comment by TrustyJAID** [comment]
I put this up because it happened on my public bot when I tested the issue posted in #coding in the Red Server. After making the issue I looked closer at the traceback and realized it was a mistake in a custom check I wrote to allow a certain command only in specific guilds with this decorator `@commands.check(lambda ctx: ctx.guild.id in TEAM_SERVERS.values())`. I can fix it by changing the check to first check if guild exists in the ctx `@commands.check(lambda ctx: ctx.guild is not None and ctx.guild.id in TEAM_SERVERS.values())` I don't know if it's something we can really fix or needs to be fixed and I jumped the gun slightly on this issue.
